src/components/CopyPaste/CopyPaste.ts

```diff
@@ -0,0 +1,95 @@
+import { ResultAction, IResultActionOptions } from '../ResultAction/ResultAction';
+import { ComponentOptions, l, IResultsComponentBindings, IQueryResult, StringUtils, Initialization } from 'coveo-search-ui';
+import { copyPaste } from '../../utils/icons';
+import './Strings';
+
+export interface ICopyPasteOptions extends IResultActionOptions {
+    /**
+     * The icon that the ResultAction will display.
+     * If text is provided, the button will contain that text.
+     * If the HTML of an SVG image is provided, that image will be displayed in the button.
+     */
+    icon?: string;
+
+    /**
+     * The tooltip that displays on hovering the ResultAction.
+     */
+    tooltip?: string;
+
+    /**
+     * Specifies the template that will be used for the copy to clipboard.
+     *
+     * Default value is `${title}\n${ClickUri}`.
+     */
+    template?: string;
+}
+
+export class CopyPaste extends ResultAction {
+    static ID = 'CopyPaste';
+
+    /**
+     * The possible options for _ResultAction_.
+     * @componentOptions
+     */
+    static options: ICopyPasteOptions = {
+        icon: ComponentOptions.buildStringOption({ defaultValue: copyPaste }),
+
+        tooltip: ComponentOptions.buildCustomOption(tooltip => tooltip, { defaultFunction: () => l('CopyPaste_copy') }),
+
+        template: ComponentOptions.buildStringOption({ defaultValue: '${title}\n${clickUri}' })
+    };
+
+    /**
+     * Construct a ResultAction component.
+     * @param element The HTML element bound to this component.
+     * @param options The options that can be provided to this component.
+     * @param bindings The bindings, or environment within which this component exists.
+     * @param result The result of the query in which this resultAction exists.
+     */
+    constructor(
+        public element: HTMLElement,
+        public options: ICopyPasteOptions,
+        public bindings?: IResultsComponentBindings,
+        public result?: IQueryResult
+    ) {
+        super(element, ComponentOptions.initComponentOptions(element, CopyPaste, options), bindings, result);
+
+        super.init();
+
+        this.bind.on(this.element, 'mouseleave', (event: MouseEvent) => {
+            if (event.target == this.element) {
+                this.tooltip.innerText = l('CopyPaste_copy');
+            }
+        });
+    }
+
+    protected doAction() {
+        this.usageAnalytics.logClickEvent({ name: 'copyToClipboard', type: 'Click' }, {}, this.result, this.element);
+        this.copyToClipboard(StringUtils.buildStringTemplateFromResult(this.options.template, this.result));
+    }
+
+    /**
+     * Sadly that's the only way of doing in in IE11 and in lockerservice.
```

Comment:
This comment could be removed. The `copyToClipboardIE11` function name is clear enough.

Reply:
will change the function name for `copyToClipboardFallback` so I'll keep this comment.